appeals/web/src/server/appeals/appeal-details/appellant-case/outcome-valid/outcome-valid.controller.js

```diff
@@ -1,5 +1,108 @@
+import { dayMonthYearToApiDateString } from '#lib/dates.js';
+import logger from '#lib/logger.js';
 import { objectContainsAllKeys } from '#lib/object-utilities.js';
-import { decisionValidConfirmationPage } from './outcome-valid.mapper.js';
+import { decisionValidConfirmationPage, updateValidDatePage } from './outcome-valid.mapper.js';
+import * as outcomeValidService from './outcome-valid.service.js';
+
+/** @type {import('@pins/express').RequestHandler<Response>}  */
+export const getValidDate = async (request, response) => {
+	renderValidDatePage(request, response);
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postValidDate = async (request, response) => {
+	const { errors, body } = request;
+
+	if (errors) {
+		return renderValidDatePage(request, response);
+	}
+
+	try {
+		if (!objectContainsAllKeys(request.session, 'appealId')) {
+			return response.render('app/500.njk');
+		}
+
+		const updatedValidDateDay = parseInt(body['due-date-day'], 10);
+		const updatedValidDateMonth = parseInt(body['due-date-month'], 10);
+		const updatedValidDateYear = parseInt(body['due-date-year'], 10);
+
+		if (
+			Number.isNaN(updatedValidDateDay) ||
+			Number.isNaN(updatedValidDateMonth) ||
+			Number.isNaN(updatedValidDateYear)
+		) {
+			return response.render('app/500.njk');
```

Comment:
I am confused if the date isn't a valid number it renders a 500 rather than error renderValidDatePage?